hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -822,17 +822,14 @@ private void validateSetOptimizeQueriesOption(boolean optimizeQueries) {
         if (!setCacheDeserializedValuesExplicitlyInvoked) {
             return;
         }
-        if (cacheDeserializedValues == CacheDeserializedValues.INDEX_ONLY) {
-            throw new ConfigurationException("Deprecated option 'optimize-queries' is being set"
-                    + ", but 'cacheDeserializedValues' was set to INDEX_ONLY. "
-                    + "These are conflicting options. Please remove the `optimize-queries'");
-        } else if (cacheDeserializedValues == CacheDeserializedValues.NEVER && optimizeQueries) {
+        if ((cacheDeserializedValues == CacheDeserializedValues.INDEX_ONLY ||
+                cacheDeserializedValues == CacheDeserializedValues.NEVER) && optimizeQueries) {
```

Comment:
really minor: for symmetry with `validateCacheDeserializedValuesOption`, maybe use: `if (cacheDeserializedValues != CacheDeserializedValues.ALWAYS)`?